### PR TITLE
Add a ResourceLinks model

### PR DESCRIPTION
**Pull request by egiurleo**
I realized that innovative policy idea and honorable mention objects are essentially `Link`s -- they link to an external URL and are associated with subcategories.

I want to create three types of `Link`s using single table inheritance. The first is the traditional link, which I'm calling a `ResourceLink`. A subcategory can have any number of resource links, and we tell which ones are still being used based on their `active` field.

Two more links, `HonorableMention`s and `InnovativePolicyIdea`s will come in a follow-up PR. These will behave differently than `ResourceLink`s, which is why I'm making them separate classes.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -20,7 +20,15 @@
     update_or_create_criterion,
     update_or_create_link,
 )
-from models import Category, Subcategory, Criterion, Link, Score, State  # noqa: E402
+from models import (  # noqa: E402, F401
+    Category,
+    Subcategory,
+    Criterion,
+    Link,
+    ResourceLink,
```
Comment (brittanylau) [inline app.py:28]: Is this import being used?

**Review by brittanylau — APPROVED** [review]
This is so cool!!! 🤩 

Will we have to update the /links endpoints use the new resource links in the future?

**Comment by egiurleo** [comment]
@brittanylau Good call! I made #105 so we remember to update the API once CMS work resumes.

File: app.py (hunk 1 above)
Comment (egiurleo) [reply app.py:28]: Nope :) I'll get rid of it.

**Pull request merged**